documentation/latency-debugging-guide.md

```diff
@@ -0,0 +1,257 @@
+executed by the JDBC driver. One statement is normally translated to one RPC call. However, there
+can be additional RPC calls being executed in the following cases:
+1. The JDBC driver inlines the `BeginTransaction` call with the first statement in a transaction.
+   This is not possible in specific cases, such as if a transaction only contains mutations, or if
+   the first statement in a transaction returns an error. In those cases, the JDBC driver executes
+   and additional `BeginTransaction` RPC.
+2. The JDBC driver retries RPCs that return `UNAVAILABLE` errors. These retries are transparent to
+   the caller, and the observed behavior from the outside is that the statement execution took a
+   longer time than expected. Enable API tracing to get insights into whether an RPC was retried or
+   not.
+
+You can enable API tracing by adding the property `enableApiTracing=true` to the JDBC connection URL.
+
+Tip: You can also enable API tracing by setting the environment variable `SPANNER_ENABLE_API_TRACING=true`
+
+### Example Using Global OpenTelemetry
+Create and register a global OpenTelemetry object before creating a JDBC connection.
+See also the [Spring Data JDBC Sample](../samples/spring-data-jdbc) for an example for how to
+configure OpenTelemetry in combination with Spring Data.
+
+```java
+TraceConfiguration traceConfiguration = TraceConfiguration.builder().setProjectId("my-project").build();
+SpanExporter traceExporter = TraceExporter.createWithConfiguration(traceConfiguration);
+OpenTelemetry openTelemetry =
+    OpenTelemetrySdk.builder()
+        .setTracerProvider(
+            SdkTracerProvider.builder()
+                .setSampler(Sampler.traceIdRatioBased(0.05))
+                .setResource(
+                    Resource.builder()
+                        .put("service.name", "my-unique-service-name")
+                        .build())
+                .addSpanProcessor(BatchSpanProcessor.builder(traceExporter).build())
+                .build())
+        .buildAndRegisterGlobal();
+String projectId = "my-project";
+String instanceId = "my-instance";
+String databaseId = "my-database";
+// Setting this to true instructs the JDBC driver to include the SQL statement with the traces.
+boolean enableExtendedTracing = true;
+
+try (Connection connection =
+   DriverManager.getConnection(
+       String.format(
+           "jdbc:cloudspanner:/projects/%s/instances/%s/databases/%s?enableExtendedTracing=%s",
+           projectId, instanceId, databaseId, enableExtendedTracing))) {
+ try (Statement statement = connection.createStatement()) {
+   try (ResultSet rs = statement.executeQuery("SELECT CURRENT_TIMESTAMP()")) {
+     while (rs.next()) {
+       System.out.printf(
+           "Connected to Cloud Spanner at [%s]%n", rs.getTimestamp(1).toString());
+     }
+   }
+ }
+}
+```
+
+### Example Using an OpenTelemetry instance in Properties
+Create and register a global OpenTelemetry object before creating a JDBC connection.
```

Comment:
Here as well. global OT is not used here

Reply:
Removed